pkg/operator/operator.go

```diff
@@ -408,9 +409,23 @@ func (o *Operator) DeployApp(appID string, sequence int64) (deployed bool, deplo
 	}
 	deployed, err = o.client.DeployApp(deployArgs)
 	if err != nil {
+		go func() {
+			logger.Info("app deploy failed, starting cluster upgrade in the background")
+			err2 := embeddedcluster.MaybeStartClusterUpgrade(context.Background(), o.k8sClientset, o.store, kotsKinds.EmbeddedClusterConfig)
+			if err2 != nil {
+				logger.Error(errors.Wrap(err2, "failed to start cluster upgrade"))
+			}
+			logger.Info("cluster upgrade started")
```

Comment:
would these logs show even if it wasn't an embedded cluster?

Reply:
Yes, I'll fix that

Reply:
updated to put this within an isEmbeddedCluster check